test/CrowdsaleGenTest.js

```diff
@@ -408,6 +415,32 @@ contract('LifCrowdsale Property-based test', function(accounts) {
     return state;
   };
 
+  let runClaimEthCommand = async (command, state) => {
+
+    let crowdsale = state.crowdsaleData,
+      { publicPresaleStartBlock, maxPresaleWei, privatePresaleRate } = crowdsale,
+      nextBlock = web3.eth.blockNumber + 1,
+      account = accounts[command.fromAccount],
+      purchases = _.filter(state.purchases, (p) => p.account == command.fromAccount);
+
+    let shouldThrow = !state.crowdsaleFinalized ||
+      !state.funded ||
+      (purchases.length == 0) ||
```

Comment:
Do we need this condition really? if there are no purchases its not funded. right? thats what we expect.

Reply:
that's what should happen, yes... but not bad to double check, there could be bugs on the funded state for example, or the other way around. For example there's one bug here: it should be `crowdsaleFunded` instead of `funded`

Reply:
:+1: